src/main/kotlin/org/dif/peerdid/PeerDIDCreator.kt

```diff
@@ -16,22 +21,46 @@ fun isPeerDID(peerDID: String): Boolean {
     return regex.matches(peerDID)
 }
 
-/** Creates [PeerDID] according to zero algorithm using [inceptionKey]*/
+/**
+ * Generates PeerDID according to the zero algorithm
+ * For this type of algorithm DIDDoc can be obtained from PeerDID
+ * @param [inceptionKey] the key that creates the DID and authenticates when exchanging it with the first peer
+ * @throws IllegalArgumentException if the [inceptionKey] is not correctly encoded
+ * @return generated PeerDID
+ */
 fun createPeerDIDNumalgo0(inceptionKey: PublicKeyAuthentication): PeerDID {
+    if (!checkKeyCorrectlyEncoded(inceptionKey.encodedValue, inceptionKey.encodingType))
+        throw IllegalArgumentException("Inception key $inceptionKey is not correctly encoded")
     return "did:peer:0".plus(createEncnumbasis(inceptionKey))
 }
 
-/** Creates [PeerDID] according to the second algorithm using [encryptionKeys], [signingKeys], [service]*/
+/**
+ * Generates PeerDID according to the second algorithm
+ * For this type of algorithm DIDDoc can be obtained from PeerDID
+ * @param [encryptionKeys] list of encryption keys
+ * @param [signingKeys] list of signing keys
+ * @param [service] JSON string conforming to the DID specification
+ * @throws IllegalArgumentException if at least one of keys is not properly encoded
```

Comment:
We throw that exception if service is not a valid JSON, right? So, let's mention this in the docs.